pkg/operator/status.go

```
@@ -86,37 +87,44 @@ func isDeploymentStatusAvailableAndUpdated(deploy appsv1.Deployment) bool {
+}
+
+// isDeploymentUpToDate returns true when the deployment controller has
+// observed the current spec and all replicas use the current pod template.
+func isDeploymentUpToDate(deploy appsv1.Deployment) bool {
+	desiredReplicas := ptr.Deref(deploy.Spec.Replicas, 1)
+	return deploy.Status.ObservedGeneration >= deploy.Generation &&
+		deploy.Status.UpdatedReplicas == desiredReplicas &&
+		deploy.Status.Replicas == desiredReplicas
 }
 
 func (c *serviceCAOperator) syncStatus(operatorConfigCopy *operatorv1.ServiceCA, existingDeployments *appsv1.DeploymentList, targetDeploymentNames sets.String) {
```

golangci-lint (2.11.4)

[error] SA1019: sets.String is deprecated: use generic Set instead. new ways: s1 := Set[string]{} s2 := New[string]() 

(staticcheck)